### PR TITLE
pangea-sdk: add Audit export API (GEA-13176)

### PR DESCRIPTION
**Pull request by kenany**

### CODE REVIEW

File: packages/pangea-sdk/pangea/request.py [hunk 1]
```diff
@@ -5,20 +5,22 @@
...
+from typing_extensions import TypeVar
 
 import pangea
 import pangea.exceptions as pe
 from pangea.config import PangeaConfig
 from pangea.response import AttachedFile, PangeaResponse, PangeaResponseResult, ResponseStatus, TransferMethod
 from pangea.utils import default_encoder
 
+if TYPE_CHECKING:
+    import aiohttp
```
Comment (pangea-andrest) [inline packages/pangea-sdk/pangea/request.py:22]: If you do not import this package won't it fail in all the lines it's called? Shouldn't you add this check (`if TYPE_CHECKING` ) in all those lines too?

File: packages/pangea-sdk/pangea/services/audit/models.py [hunk 2]
```diff
@@ -441,7 +441,10 @@ def __repr__(self):
 
 
 class DownloadRequest(APIRequestModel):
-    result_id: str
+    request_id: Optional[str] = None
+    """ID returned by the search API."""
+
+    result_id: Optional[str] = None
     """ID returned by the search API."""
```
Comment (pangea-andrest) [inline packages/pangea-sdk/pangea/services/audit/models.py:448]: Should they have different descriptions?

File: packages/pangea-sdk/tests/integration/asyncio/test_audit.py [hunk 3]
```diff
@@ -758,3 +758,24 @@ async def test_download(self):
 
         file = await self.audit_general.download_file(url=response_download.result.dest_url)
         file.save("./")
+
+    async def test_export_download(self) -> None:
+        export_res = await self.audit_general.export(verbose=False)
+        self.assertEqual(export_res.status, "Accepted")
+
+        # Note that the export can easily take dozens of minutes, if not longer,
```
Comment (pangea-andrest) [inline packages/pangea-sdk/tests/integration/asyncio/test_audit.py:766]: Should do same retry logic here?

File: packages/pangea-sdk/tests/integration/test_audit.py [hunk 4]
```diff
@@ -758,6 +758,27 @@ def test_download(self):
         file = self.audit_general.download_file(url=response_download.result.dest_url)
         file.save(dest_folder="./")
 
+    def test_export_download(self) -> None:
+        export_res = self.audit_general.export(verbose=False)
+        self.assertEqual(export_res.status, "Accepted")
+
+        # Note that the export can easily take dozens of minutes, if not longer,
```
Comment (pangea-andrest) [inline packages/pangea-sdk/tests/integration/test_audit.py:765]: Should do same retry logic here?

**Review by pangea-andrest — APPROVED** [review]

File: packages/pangea-sdk/pangea/request.py (hunk 1 above)
Comment (kenany) [reply packages/pangea-sdk/pangea/request.py:22]: That not necessary, no.

File: packages/pangea-sdk/pangea/services/audit/models.py (hunk 2 above)
Comment (kenany) [reply packages/pangea-sdk/pangea/services/audit/models.py:448]: They should but I promise when I wrote these, that's how they were in the API docs. Now I see the docs have been updated (it now clarifies which comes from the search API and which comes from the export API), so I've updated this too now.

File: packages/pangea-sdk/tests/integration/asyncio/test_audit.py (hunk 3 above)
Comment (kenany) [reply packages/pangea-sdk/tests/integration/asyncio/test_audit.py]: Done.

File: packages/pangea-sdk/tests/integration/test_audit.py (hunk 4 above)
Comment (kenany) [reply packages/pangea-sdk/tests/integration/test_audit.py]: Done.

**Review by pangea-andrest — APPROVED** [review]

**Pull request merged**